File: 78-Subsets.cpp
class Solution {
public:
    vector<int> v;
    void ss(int i, vector<int>& p, vector<vector<int>>& res){
        if(i==v.size()){
            res.push_back(p);
            for(const auto&  e : p){cout<<e<<" ";}
            cout<<endl;
            return;
        }else{
            ss(i+1, p, res);
            p.push_back(v[i]);
            ss(i+1, p, res);
        }
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        v = nums;
        vector<vector<int>> res;
        vector<int> p;
        ss(0, p, res);
        for(const auto& r : res){
            for(const auto&  e : r){
                cout<<e<<" ";
            }
            cout<<endl;
        }
        return res;
    }
};

File: 78-Subsets.py
class Solution:
    def ss(self, nums, index, whatever, another_whatever):
        if(index==len(nums)):
            print(index, whatever)
            #aaw = whatever.copy()
            #another_whatever.append(aaw)
            another_whatever.append(whatever.deepcopy())
            return
        else:
            self.ss(nums,index+1,whatever,another_whatever)
            whatever.append(nums[index])
            self.ss(nums,index+1,whatever,another_whatever)
            del whatever[-1]
        
    def subsets(self, nums: List[int]) -> List[List[int]]:
        whatever = []
        another_whatever = []
        self.ss(nums,0,whatever,another_whatever)
        return another_whatever
